components/ledger/internal/services/command/sync-balances-batch.go

```
@@ -42,7 +43,7 @@ type SyncBalancesBatchResult struct {
 //   - Missing keys (already expired): skipped in aggregation
 //   - Version conflicts: optimistic locking in DB update
 //   - Partial failures: keys only removed after successful DB write
-func (uc *UseCase) SyncBalancesBatch(ctx context.Context, organizationID, ledgerID uuid.UUID, keys []string) (*SyncBalancesBatchResult, error) {
+func (uc *UseCase) SyncBalancesBatch(ctx context.Context, organizationID, ledgerID uuid.UUID, keys []redisTransaction.SyncKey) (*SyncBalancesBatchResult, error) {
```

golangci-lint (2.11.4)

[error] cognitive complexity 45 of func `(*UseCase).SyncBalancesBatch` is high (> 30)

(gocognit)